rama-http-types/src/curl/mod.rs

```diff
@@ -0,0 +1,45 @@
+use http::{HeaderMap, HeaderValue, Method, Request, Uri, Version};
+use rama_error::OpaqueError;
+
+// TODO: make sealed(private) trait, and add sealed(private) impl for Request<T>
+pub trait IntoCurlHeadersCommand {
+    fn to_curl_command(&self) -> Result<String, OpaqueError>;
```

Comment:
I'm fine in moving this to a sealed trait, but in that case you also want to move this method to that trait, such that the public trait becomes fully opaque